### DISCUSSION TITLE
0.25 release announcement

### DISCUSSION THREAD

**Discussion by Shnatsel** [image-rs/image#2204] [General]
I finally sat down and hammered out a draft of the release announcement: https://hackmd.io/@xmxSgN6MSImJPcY4MOKfjA/rkS2q9Be0

@fintelia I would appreciate if you could take a look. There are some TODOs that I don't think I can fill in myself - they require a deeper knowledge of `image` than what I possess.

**Comment by fintelia** [comment]
Thanks for writing this! I left a comment explaining the motivation for the API change. The get involved part is a bit harder, because one of the big bottlenecks is PR reviewer bandwidth which isn't something new contributors can necessarily help with

**Comment by Shnatsel** [comment]
@fintelia I've integrated your comment and added a lengthy "Get involved" bit. Please take another look at your leisure.

**Reply by 197g** [reply]
In "Get involved" I would list the two major under-implement categories which get requested most frequently: support for metadata consistent with exif, treatment of color-spaces during de/encoding. In contrast to performance, always relevant and it can be done incremental, those two missing interfaces probaby divert actual a few use-cases and thus available programming effort away.

**Reply by Shnatsel** [reply]
> support for metadata consistent with exif, treatment of color-spaces during de/encoding

I do not quite understand what those things involve, so I cannot explain them in an announcement either. It would be great to have issues on the tracker with detailed descriptions of what needs to be done that I could link to

**Reply by fintelia** [reply]
I wrote up some thoughts in https://github.com/image-rs/image/issues/2222 on what might be required for exif + color spaces, but there's still a lot of design work left (and then a far larger amount of implementation work!). I'm somewhat hesitant to highlight that area in the help wanted post because it seems better to focus on "good first issue" sorts of things rather than "so big and complicated even the maintainers haven't dared attempt it" features

**Reply by 197g** [reply]
Agreed, both are too large topic for incidental contributions. Those are precisely the kind of work where it might be necessary to use or commit company resources for, fitting with the last paragraph. But yes, ultimately it might be best not to overcomplicate the release announcement with those topics since some summary of existing and required efforts would be needed.

**Comment by fintelia** [comment]
At this point the release announcement looks good to me!

**Reply by Shnatsel** [reply]
I forgot to add GIF changes: https://github.com/image-rs/image-gif/blob/master/Changes.md

Many of them are quite exciting as well. I'll add a section about GIF changes too.

**Reply by Shnatsel** [reply]
Done. I've added a section about GIF.

**Comment by Shnatsel** [comment]
Posted: https://redd.it/1cj94va
